packages/date-time-picker/test/date-time-picker.test.ts

```diff
@@ -10,30 +10,1908 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+
+                    expect(hourSegment.innerText).to.equal(`00`);
+                    expectPlaceholders(editableSegments, [hourSegment]);
+                });
+
+                it("incrementing the hour segment's value ", async () => {
+                    hourSegment.focus();
+                    await sendKeyMultipleTimes('ArrowUp', 14);
+                    await elementUpdated(element);
+
+                    expect(hourSegment.innerText).to.equal(`13`);
+                    expectPlaceholders(editableSegments, [hourSegment]);
+                });
+
+                it('resetting the hour when the max is reached', async () => {
+                    hourSegment.focus();
+                    await sendKeyMultipleTimes('ArrowUp', 25);
+                    await elementUpdated(element);
+
+                    expect(hourSegment.innerText).to.equal(`00`);
+                    expectPlaceholders(editableSegments, [hourSegment]);
+                });
+            });
+
+            describe('updating the day', () => {
+                let daySegment: HTMLElement;
+                let monthSegment: HTMLElement;
+                let yearSegment: HTMLElement;
+
+                beforeEach(async () => {
+                    daySegment = editableSegments.getByType(SegmentTypes.Day);
+                    monthSegment = editableSegments.getByType(
+                        SegmentTypes.Month
+                    );
+                    yearSegment = editableSegments.getByType(SegmentTypes.Year);
+
+                    daySegment.focus();
+                    await sendKeys({ press: 'ArrowDown' });
+                });
+
+                it('when the month changes to February with no year selected', async () => {
+                    monthSegment.focus();
+                    await sendKeys({ press: 'ArrowUp' });
+                    await sendKeys({ press: 'ArrowUp' });
+                    await elementUpdated(element);
+
+                    expect(monthSegment.innerText).to.equal(`02`);
+                    expect(daySegment.innerText).to.equal(`29`);
+                    expectPlaceholders(editableSegments, [
+                        daySegment,
+                        monthSegment,
+                    ]);
+                });
+
+                it('when the month changes to February in a common year', async () => {
+                    yearSegment.focus();
+                    await sendKeys({ press: 'ArrowUp' });
+                    await elementUpdated(element);
+                    monthSegment.focus();
+                    await sendKeys({ press: 'ArrowUp' });
+                    await sendKeys({ press: 'ArrowUp' });
+                    await elementUpdated(element);
+
+                    expect(yearSegment.innerText).to.equal(`2022`); // Common year in the Gregorian calendar
+                    expect(monthSegment.innerText).to.equal(`02`);
+                    expect(daySegment.innerText).to.equal(`28`);
```

Comment:
For incrementing/ decrementing/ typing in/ clearing the segments when a date is defined (the DateTimePicker has a value), the tests should also check for a correct update of said value. These will be added in a follow-up.